packages/core/content-manager/server/controllers/collection-types.js

```diff
@@ -181,6 +181,70 @@ module.exports = {
     ctx.body = await permissionChecker.sanitizeOutput(result);
   },
 
+  async bulkPublish(ctx) {
+    const { userAbility } = ctx.state;
+    const { model } = ctx.params;
+    const { body } = ctx.request;
+    const { ids } = body;
+
+    await validateBulkActionInput(body);
+
+    const entityManager = getService('entity-manager');
+    const permissionChecker = getService('permission-checker').create({ userAbility, model });
+
+    if (permissionChecker.cannot.publish()) {
+      return ctx.forbidden();
+    }
+
+    const entityPromises = ids.map((id) => entityManager.findOneWithCreatorRoles(id, model));
+    const entities = await Promise.all(entityPromises);
+
+    for (const entity of entities) {
+      if (!entity) {
+        return ctx.notFound();
+      }
+
+      if (permissionChecker.cannot.publish(entity)) {
+        return ctx.forbidden();
+      }
+    }
+
+    const { count } = await entityManager.publishMany(entities, model);
+    ctx.body = { count };
+  },
+
+  async bulkUnpublish(ctx) {
+    const { userAbility } = ctx.state;
+    const { model } = ctx.params;
+    const { body } = ctx.request;
+    const { ids } = body;
+
+    await validateBulkActionInput(body);
+
+    const entityManager = getService('entity-manager');
+    const permissionChecker = getService('permission-checker').create({ userAbility, model });
+
+    if (permissionChecker.cannot.unpublish()) {
+      return ctx.forbidden();
+    }
```

Comment:
Why do we have an unpublish permission in the backend, when in the admin it looks like publish and unpublish are the same action?

Reply:
https://github.com/strapi/strapi/blob/13502afca039cc21e1caa9553f13e4816419f200/packages/core/content-manager/server/services/permission-checker.js#L5-L12

It looks like they are actually the same but I guess it is this way in the permissionChecker for readability 🤷‍♂️ . 

In the routes we reference only a 'publish' permission:

https://github.com/strapi/strapi/blob/13502afca039cc21e1caa9553f13e4816419f200/packages/core/content-manager/server/routes/admin.js#L157-L188